src/TEdit/ViewModel/WorldViewModel.cs

```diff
@@ -1046,7 +1046,7 @@ private void SaveWorldAs()
                 {
                     try
                     {
-                        var name = sfd.Filter.Split('|')?[sfd.FilterIndex]?.Split(' ').LastOrDefault()?.Trim(' ', 'v', '.');
+                        var name = sfd.Filter.ToString().Replace("Terraria World File|", "").Replace("*.wld|Terraria v", "").Replace("*.wld", "").Split('|')[sfd.FilterIndex-2];

```

Comment:
are you sure about this `-2`?

Reply:
Yes. I have tested this and the -2 is required for selecting the correct array item.